aten/src/ATen/native/cpu/UpSampleKernel.cpp

```diff
@@ -764,14 +764,19 @@ struct HelperInterpBase {
       wt_ptr[j] = w;
       total_w += w;
     }
-    for (j = 0; j < xsize; j++) {
-      if (total_w != 0.0) {
+
+    scalar_t wt_max = 0.0;
+    if (total_w != 0.0) {
+      for (j = 0; j < xsize; j++) {
         wt_ptr[j] /= total_w;
+        wt_max = std::max(wt_max, wt_ptr[j]);
       }
     }
+
     for (; j < max_interp_size; j++) {
       wt_ptr[j] = static_cast<scalar_t>(0.0);
     }
+    return wt_max;
```

Comment:
Previously, for `uint8` we were doing 3 passes over `interp_size * output_size` array of weights:
1) computing `wt_ptr` with `filter_fn`
2) normalizing `wt_ptr` with `total_w`
3) computing weights max

Now, we are doing 2 passes for all dtypes where step 2 and 3 are unified. On modern CPUs I do not think it would change much the perfs when we execute two ops (`wt_ptr[j] /= total_w;` and `wt_max = std::max(wt_max, wt_ptr[j]);`) in the loop vs one op. 

I did explicit measurements previously on that, for float32 dtype to check the overhead. Conclusion: no overhead.
- Before, `wt_max` is NOT computed inside `_compute_weights_aa`:
```
Num threads: 1

PIL version:  9.0.0.post1
[------------------------------------------------ Resize ------------------------------------------------]
                                                                          |  torch (2.1.0a0+git0c58e8a) PR
1 threads: -----------------------------------------------------------------------------------------------
      3 torch.float32 channels_first bilinear 700 -> (224, 224) aa=True   |              2586.9
      4 torch.float32 channels_first bilinear 700 -> (224, 224) aa=True   |              3437.1
      3 torch.float32 channels_first bilinear 700 -> (224, 224) aa=False  |              1832.3
      4 torch.float32 channels_first bilinear 700 -> (224, 224) aa=False  |               996.8
      3 torch.float32 channels_first bilinear 520 -> (224, 224) aa=True   |              1657.1
      4 torch.float32 channels_first bilinear 520 -> (224, 224) aa=True   |              2195.9
      3 torch.float32 channels_first bilinear 520 -> (224, 224) aa=False  |              1416.8
      4 torch.float32 channels_first bilinear 520 -> (224, 224) aa=False  |               988.3
      3 torch.float32 channels_first bilinear 256 -> (224, 224) aa=True   |               621.0
      4 torch.float32 channels_first bilinear 256 -> (224, 224) aa=True   |               816.9
      3 torch.float32 channels_first bilinear 256 -> (224, 224) aa=False  |              1049.6
      4 torch.float32 channels_first bilinear 256 -> (224, 224) aa=False  |               967.8

Times are in microseconds (us).
```
and 
- After, `wt_max` is computed inside `_compute_weights_aa`:
```
Num threads: 1

PIL version:  9.0.0.post1
[------------------------------------------------ Resize ------------------------------------------------]
                                                                          |  torch (2.1.0a0+git0c58e8a) PR
1 threads: -----------------------------------------------------------------------------------------------
      3 torch.float32 channels_first bilinear 700 -> (224, 224) aa=True   |              2586.5
      4 torch.float32 channels_first bilinear 700 -> (224, 224) aa=True   |              3460.0
      3 torch.float32 channels_first bilinear 700 -> (224, 224) aa=False  |              1823.1
      4 torch.float32 channels_first bilinear 700 -> (224, 224) aa=False  |               986.2
      3 torch.float32 channels_first bilinear 520 -> (224, 224) aa=True   |              1668.1
      4 torch.float32 channels_first bilinear 520 -> (224, 224) aa=True   |              2207.0
      3 torch.float32 channels_first bilinear 520 -> (224, 224) aa=False  |              1415.0
      4 torch.float32 channels_first bilinear 520 -> (224, 224) aa=False  |               975.7
      3 torch.float32 channels_first bilinear 256 -> (224, 224) aa=True   |               624.4
      4 torch.float32 channels_first bilinear 256 -> (224, 224) aa=True   |               820.1
      3 torch.float32 channels_first bilinear 256 -> (224, 224) aa=False  |              1048.9
      4 torch.float32 channels_first bilinear 256 -> (224, 224) aa=False  |               955.9

Times are in microseconds (us).
```